src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -38,15 +39,15 @@ private Solution(SolutionState state)
         }
 
         internal Solution(Workspace workspace, SolutionInfo.SolutionAttributes solutionAttributes, SolutionOptionSet options, IReadOnlyList<AnalyzerReference> analyzerReferences)
-            : this(new SolutionState(workspace.PrimaryBranchId, new SolutionServices(workspace), solutionAttributes, options, analyzerReferences))
+            : this(new SolutionState(workspace.PrimaryBranchId, workspace.Services, solutionAttributes, options, analyzerReferences))
         {
         }
 
         internal SolutionState State => _state;
 
         internal int WorkspaceVersion => _state.WorkspaceVersion;
 
-        internal SolutionServices Services => _state.Services;
+        public HostWorkspaceServices Services => _state.Services;
```

Comment:
Is this intended to be public? Need API review approval if so.

Reply:
yes.  this will go through API review.

Reply:
API review tracked through https://github.com/dotnet/roslyn/issues/62914